knowledge_engine/knowledge_engine/ingestion/qa_unitizer.py

```
@@ -0,0 +1,232 @@
+from typing import Any
+
+from llama_index.core import Document
+from llama_index.core.schema import TextNode
+
+MIN_QA_PAIRS_FOR_DOCUMENT = 2
+MIN_QA_COVERAGE = 0.35
+
+QUESTION_LINE_RE = re.compile(
+    r"^\s*(?:[-*]\s*)?(?:\*\*)?"
+    r"(?P<label>(?:Q(?:uestion)?|问题)\s*(?P<number>\d+)?)"
+    r"\s*[：:.)、]\s*(?P<question>.+?)"
```

Ruff (0.15.20)

[warning] String contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF001)